File: MyLanguageUtils/src/main/java/ru/herooo/mylanguageutils/file/filesize/FileSize.java
package ru.herooo.mylanguageutils.file.filesize;

public class FileSize {
    private FileSizeUnits fileSizeUnits;
    private long size;

    public FileSize(FileSizeUnits fileSizeUnits, long size) {
        this.fileSizeUnits = fileSizeUnits;
        this.size = size;
    }

    public String createSizeStr() {
        return String.format("%d%s", size, fileSizeUnits.TITLE);
    }

    public long getNumberOfBytes() {
        return size * fileSizeUnits.NUMBER_OF_BYTES;
    }
}

File: MyLanguageWeb/src/main/resources/static/js/classes/html/input/file/mime_type_with_size/file_size/file_size.js
export class FileSize {
    #size;
    #fileSizeUnit;

    constructor(size, fileSizeUnitObj) {
        this.#size = size;
        this.#fileSizeUnit = fileSizeUnitObj;
    }

    createStr() {
        let str;

        let size = this.#size;
        let fileSizeUnit = this.#fileSizeUnit;
        if (size && fileSizeUnit) {
            str = `${size}${fileSizeUnit.TITLE}`;
        }

        return str;
    }

    getNumberOfBytes() {
        let numberOfBytes = 0;

        let size = this.#size;
        let fileSizeUnit = this.#fileSizeUnit;
        if (size && fileSizeUnit) {
            numberOfBytes = size * fileSizeUnit.NUMBER_OF_BYTES;
        }

        return numberOfBytes;
    }
}